zulip/zulip
# Improve live update in the navbar

This significantly improves the live update in the navbar. I suggest having a close look at the changes to filter.js, I don't know if we're be happy with that pattern, even if it works.

**Testing Plan:** 
![live update in navbar for privacy and name](https://user-images.githubusercontent.com/33805964/84208533-9735e580-aad1-11ea-8e01-6b6565aa0fb8.gif)

Contrary to what the commit says, this does fix the live update for the stream name... That was a little unexpected... as a result of this there are a few minor todos here:
1. See if we can get the stream color to work.
2. Rewrite commit to reflect that we do change the stream name.
3. Add Fixes: #14728.

**GIFs or Screenshots:**
